content/en/docs/concepts/architecture/nodes.md

```diff
@@ -513,6 +513,33 @@ During a non-graceful shutdown, Pods are terminated in the two phases:
   recovered since the user was the one who originally added the taint.
 {{< /note >}}
 
+### Forced storage detach on timeout {#storage-force-detach-on-timeout}
+
+In any situation where a pod deletion has not succeeded for 6 minutes, kubernetes will
+force detach unmounting volumes if the node is unhealthy at that instant. Any
+workload still running on the node that uses a force-detached volume will cause a
+violation of the
+[CSI specification](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerunpublishvolume),
+which states that ControllerUnpublishVolume "MUST be called after all
+NodeUnstageVolume and NodeUnpublishVolume on the volume are called and succeed".
+In such circumstances, volumes on the node in question might encounter data corruption.
+
+With the advent of the "Non-graceful node shutdown" feature, users might opt to disable
+this behavior.
+
+Force storage detach on timeout can be disabled by setting the `disable-force-detach-on-timeout`
+flag in `kube controller manager`; this means that a volume that is hosted on a node that
+is unhealthy for more than 6 minutes will not have its associated
+[Volume Attachment](/docs/reference/kubernetes-api/config-and-storage-resources/volume-attachment-v1/)
+deleted.
+
+However, manual mitigations might be required to repair the pod in question:
+* The user might opt to manually taint the underlying node via the
+[Non-Graceful Node Shutdown](#non-graceful-node-shutdown) procedure mentioned above if the node is
+truly detached.
+* If using the Non-Graceful Node Shutdown procedure is not an option, the user the user should
+opt to reboot the node in question.
```

Comment:
I don't like presenting these as 2 alternatives, because the recommended mitigation is probably some combination of both. A reboot that brings the node back quickly is one option, but the more realistic option is to both shut down the node and apply the taint, then try to bring the node back.

I am very sensitive to any language which suggests that it's okay to apply the taint without actually shutting the node down, because that can still lead the problems. That misunderstanding can cause data corruption by itself, so we should reinforce the idea that the taint can only be applied after shutting down the node.

I propose this language:

There are two recommended options for cleaning up an unhealthy pod which is attached to a volume:
* Ensure the node is shut down, then taint the node according to the
[Non-Graceful Node Shutdown](#non-graceful-node-shutdown) procedure mentioned above.
* Reboot the node and wait for it to rejoin the cluster as a healthy node.

Reply:
> There are two recommended options for cleaning up an unhealthy pod which is attached to a volume:

If we're not giving the user "alternatives", then I don't think we should be giving them "options" either. Moreover, I think the steps are probably adequately captured in the non-graceful node shutdown above.

I think I'll just add a pointer to that section and indicate that that's the only way to fix the pod.